src/pages/index.js

```diff
@@ -87,25 +119,92 @@ const addFormValidator = new FormValidator(
   cardForm
 );
 
+const avatarFormValidator = new FormValidator(
+  constants.validationSettings,
+  avatarForm
+);
+
 const profileInfo = new UserInfo({
   name: ".profile__name",
   description: ".profile__description",
 });
 
-const editProfilePopup = new PopupWithForm("#edit-popup", handleSaveProfile);
-const newCardPopup = new PopupWithForm("#new-card-popup", handleAddCard);
+const editProfilePopup = new PopupWithForm("#edit-popup", handleUpdateProfile);
+
+const updateAvatarPopup = new PopupWithForm(
+  "#avatar-popup",
+  handleUpdateAvatar
+);
+
 const imagePopup = new PopupWithImage("#preview-image-popup");
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                Gallery Creation                                ||
 // ! ||--------------------------------------------------------------------------------||
 
-const cardGallery = new Section(
-  { items: constants.initialCards, renderer: renderCard },
-  ".gallery__cards"
-);
-
-cardGallery.renderItems();
+api.getInitialCards().then((cards) => {
+  const cardGallery = new Section(
+    { items: cards, renderer: renderCard },
+    ".gallery__cards"
+  );
+
+  function renderCard(cardData) {
+    const card = new Card(
+      {
+        cardData,
+
+        handleImageClick,
+        handleDeleteButton,
+        handleCardLike,
+        handleCardUnlike,
+      },
+      cardTemplate
+    );
+
+    function handleDeleteButton() {
+      console.log(document.querySelector(`.modal_type_delete`));
+      deletePopup.open();
+    }
+
+    function handleCardLike(id) {
+      api.likeCard(id);
+    }
+
+    function handleCardUnlike(id) {
+      api.unlikeCard(id);
+    }
+
+    const handleDeleteSubmit = () => {
+      let id = card.getId();
+      api.removeCard(id);
+      card.removeItem();
+      // For some reason, when this function is called consecutively,
+      // it saves the card's ID from each previous time it was called,
+      // so it tries to delete those IDs as well.
+    };
+
+    const deletePopup = new PopupDelete("#delete-popup", handleDeleteSubmit);
+
+    return card.getView();
+  }
+
+  function handleAddCard({ title: name, link: link }) {
+    api.addCard({ name, link }).then((data) => {
+      cardGallery.addItem(data);
+    });
+    newCardPopup.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.